spark_log_parser/loaders/https.py

```diff
@@ -29,7 +29,10 @@ def load_item(self, url):
             raise AssertionError("Download is empty")
 
         wrapped = FileChunkStreamWrapper(response.iter_content(chunk_size=self._STREAM_CHUNK_SIZE))
-        yield from self.extract(Path(url), wrapped)
+        try:
+            yield from self.extract(Path(url), wrapped)
+        finally:
+            response.close()
```

Comment:
related to the discussion about connections remaining open, just make sure we explicitly close all HTTP response streams we may have (same for S3 files). We don't need to do this for local files, since we just pass a `Path` in and not an already-open file stream. Combined with that fact that we use contextmanagers to open filepaths, those resources will auto-close themselves.

Reply:
Nice. You could also use the response as a context manager: https://requests.readthedocs.io/en/latest/user/advanced/#body-content-workflow

Reply:
Ah - that's good to know